### PR TITLE
[KOGITO-5189] Springboot persistence IT and remove persistence profile

### PR DESCRIPTION
**Pull request by tiagodolphine**
Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](CONTRIBUTING.md)
- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/master/kogito-build/kogito-ide-config)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

* <b>Full Kogito testing</b> (with cloud images and operator BDD testing)  
  Please add comment: <b>Jenkins run BDD</b>  
  <b>This check should be used only if a big change is done as it takes time to run, need resources and one full BDD tests check can be done at a time ...</b>
</details>

### CODE REVIEW

**Comment by tiagodolphine** [comment]
jenkins retest this

**Review by cristianonicolai — APPROVED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -84,22 +84,6 @@ pipeline {
                 }
             }
         }
-        stage('Check Runtimes integration-tests with persistence') {
```
Comment (danielezonca) [inline Jenkinsfile:87]: We need to remove the same step also here
https://github.com/kiegroup/kogito-pipelines/blob/master/.ci/pull-request-config.yaml#L21
https://github.com/kiegroup/kogito-pipelines/blob/master/.ci/pull-request-config.yaml#L24

Btw I can do it with another PR of necessary, I mentioned it for the record :)

**Review by danielezonca — APPROVED** [review]
LGTM, Just some minor comments :)

File: Jenkinsfile (hunk 1 above)
Comment (danielezonca) [reply Jenkinsfile:87]: PR created :) 
https://github.com/kiegroup/kogito-pipelines/pull/191
I will merge it after this PR

Comment (tiagodolphine) [reply Jenkinsfile:87]: +1 thanks

File: integration-tests/integration-tests-springboot/src/it/integration-tests-springboot-persistence-it/pom.xml [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        <configuration>
+          <classesDirectory>${project.build.outputDirectory}</classesDirectory>
+          <workingDirectory>${project.build.directory}/test-run</workingDirectory>
+
+          <useModulePath>false</useModulePath>
+          <systemPropertyVariables combine.children="append">
+            <container.image.keycloak>${container.image.keycloak}</container.image.keycloak>
+            <container.image.infinispan>${container.image.infinispan}</container.image.infinispan>
+            <container.image.kafka>${container.image.kafka}</container.image.kafka>
+            <native.image.path>${project.build.directory}/${project.build.finalName}-runner</native.image.path>
```
[the remark this answers is not in the document]
Comment (tiagodolphine) [reply integration-tests/integration-tests-springboot/src/it/integration-tests-springboot-persistence-it/pom.xml:126]: copy/paste, I'll remove it.

File: integration-tests/integration-tests-springboot/src/it/integration-tests-springboot-persistence-it/pom.xml [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+            <exclude>${tests.failsafe.include}</exclude>
+          </excludes>
+        </configuration>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-failsafe-plugin</artifactId>
+        <configuration>
+          <classesDirectory>${project.build.outputDirectory}</classesDirectory>
+          <workingDirectory>${project.build.directory}/test-run</workingDirectory>
```
[the remark this answers is not in the document]
Comment (tiagodolphine) [reply integration-tests/integration-tests-springboot/src/it/integration-tests-springboot-persistence-it/pom.xml]: only the `classesDirectory` is necessary, without it I got a bunch of errors during the spring boot DI startup, beans were not found. I cleaned a bit this pom.

**Comment by tiagodolphine** [comment]
jenkins retest this

**Review by MarianMacik — COMMENTED** [review]
LGTM.

**Pull request merged**